File: Check if string is rotated by two places/Check if string is rotated by two places.py
class Solution:
    #Function to check if a string can be obtained by rotating
    #another string by exactly 2 places.
    def isRotated(self,str1,str2):
        #code here
        n = len(str1)
        n1 = len(str2)
        if n != n1:
            return 0
        c = ''
        ac = ''
        ac = str2[n-2:n] + str2[0:n-2]
        c = str2[2:n] + str2[0:2]
        if c == str1 or ac == str1:
            return 1
        else:
            return 0

File: Check if string is rotated by two places/Check if string is rotated by two places.cpp
class Solution
{
    public:
    //Function to check if a string can be obtained by rotating
    //another string by exactly 2 places.
    bool isRotated(string str1, string str2)
    {
        // Your code here
        int n1=str1.size();
        int n2=str2.size();
        if(n1!=n2||n1<2)
        return 0;
        string s="";
        string s1="";
        for(int i=2;i<n1;i++)
        {
            s+=str1[i];
        }
        s+=str1[0];
        s+=str1[1];
        s1+=str1[n1-2];
        s1+=str1[n1-1];
        for(int i=0;i<n1-2;i++)
        {
            s1+=str1[i];
        }
        if(s==str2||s1==str2)
        {
            return 1;
        }
        return 0;
    }

};